python/ray/serve/application.py

```diff
@@ -258,6 +261,8 @@ def __setitem__(self, key: str, value: Deployment):
         elif not isinstance(value, Deployment):
             raise TypeError(f"Got {type(Deployment)} for value. Expected deployment.")
 
+        value._func_or_class = get_deployment_import_path(value)
```

Comment:
Hmm the problem is this will throw if a user defines something in `__main__` and we want them to be able to do this at least for local testing.

Can we instead only enforce this when we do `to_yaml`/`to_dict`?

Reply:
Yes, we can do that.